nemo/collections/tts/torch/helpers.py

```diff
@@ -50,14 +51,28 @@ def general_padding(item, item_len, max_len, pad_value=0):
     return item
 
 
-def beta_binomial_prior_distribution(phoneme_count, mel_count, scaling_factor=1.0):
-    x = np.arange(0, phoneme_count)
-    mel_text_probs = []
-    for i in range(1, mel_count + 1):
-        a, b = scaling_factor * i, scaling_factor * (mel_count + 1 - i)
-        mel_i_prob = betabinom(phoneme_count - 1, a, b).pmf(x)
-        mel_text_probs.append(mel_i_prob)
-    return np.array(mel_text_probs)
+def logbeta(x, y):
+    return gammaln(x) + gammaln(y) - gammaln(x + y)
+
+
+def logbetabinom(n, a, b, x):
```

Comment:
Is there a way to implement this using out-of-the-box logic in Pytorch distributions?

https://pytorch.org/docs/stable/distributions.html

Reply:
I don't immediately see a neat way to implement this using PyTorch distributions.

PyTorch doesn't have beta-binomial distribution, and to implement its pdf we need gamma and beta functions rather than corresponding distribution likelihoods.

Reply:
I also don't see any neat way to implement it. PyTorch distributions looks very limited compared to Scipy and Tensorflow probability, both in terms of functionality and interfaces.